### PR TITLE
Added convenience functions to DatabaseRecord class

### PR DESCRIPTION
**Pull request by jcgoodru**
Added several functions to DatabaseRecord class to make life easier.

**Key features:**
- Bulk write operations: no more need to use ```pymongo``` syntax to save multiple items efficiently.  This functionality is built into the ```FooRecord``` classes now.
- Easier-to-use syntax for CRUD operations: Creating, finding, and updating objects is more intuitive (IMO) through the use of factory methods

**Functions Added**
```save_all(list_of_records)``` Saves all records at once
```new(optional_dict=None)``` Creates a new record, with the values in optional_dict if wanted
```find_one(query_dict)``` Returns None, a record, or throws an error
```find_all(query_dict)``` Returns a pymongo.Cursor object with zero or more records

Example Usage (aka, what's different):
```
from database.tables.league.players import PlayerRecord

############ This is the new stuff ############

## Creates a new PlayerRecord instance (default values)
rec = PlayerRecord.new()
print rec.player_id      ## prints None
print rec.player_name    ## prints None


## Creates a new PlayerRecord instance (some custom values)
lonzo = PlayerRecord.new({ f.player_name : "Lonzo Ball" })
print lonzo.player_id      ## prints None
print lonzo.player_name    ## prints "Lonzo Ball"


## Finds one PlayerRecord
bron = PlayerRecord.find_one({ f.player_id : 2544 })
returns_none = PlayerRecord.find_one({ f.player_id : -69 })
try:
    throws_error = PlayerRecord.find_one({ f.from_year : 2011 })
except:
    print "This found more than one, which isn't allowed here"


## Finds many PlayerRecords
cursor = PlayerRecord.find_all({ f.from_year : 2011 }) ## Finds a list of people drafted in 2011


## Saves one record.  Updates it's _id and timestamps
rec.save()

## Saves many records.  Updates all of their _ids and timestamps
records = [lonzo, bron]
status = PlayerRecord.save_all(records)
print status ## prints "{ 'INSERTED' : 1, 'UPDATED' : 1 }"



############ This is the old stuff ############
############ You can still do this ############

rec = connection.PlayerRecord()
rec = connection.PlayerRecord.one()
cursor = connection.PlayerRecord.find()
rec.save()
```

Testing done:
 - Made sure none of the preexisting code broke by running backfill 2x on an empty database.  Seems like it's working just fine (wishful thinking, I know)
 - Tested the new operations manually:
        - Did not let me save without required fields (Good)
        - Did not let me save with an invalid type (ex: player_id as a float) (Good)
        - Did not let me save one record with duplicate unique keys (Good)
        - Did not let me save multiple new records when the keys are in the DB (Good)

### CODE REVIEW

**Comment by jcgoodru** [comment]
tl;dr:  You can use this:
```
new_rec = PlayerRecord.new()
old_rec = PlayerRecord.find_one({ f.player_id : 2544 })
many_old_recs = PlayerRecord.find_all({ f.player_id : 2544 })

things_to_save = get_list_of_things_to_save()
PlayerRecord.save_all(things_to_save)
```
Instead of this:
```
new_rec = connection.PlayerRecord()
old_rec = connection.PlayerRecord.one({ f.player_id : 2544 })
many_old_recs = connection.PlayerRecord.find({ f.player_id : 2544 })

things_to_save = get_list_of_things_to_save()
for rec in things_to_save:
    rec.save()
```

**Comment by RyanDanielOMara** [comment]
#176
